packages/core/workers/src/WorkerFarm.js

```diff
@@ -350,7 +341,13 @@ export default class WorkerFarm extends EventEmitter {
   }
 
   static isWorker() {
-    return process.send && require.main.filename === require.resolve('./child');
+    try {
```

Comment:
this is leaking a bit about the backend into the WorkerFarm.

I wonder if `Worker` could accept a backend as an argument and it could be determined in `WorkerFarm` instead. That way, we could also use something like `WorkerImpl.isWorker()` here. This would allow us to easily scale beyond these two backends too.

Reply:
Yeah I was thinking about that too. Would allow for additional backends like web workers as well.